tests/components/mqtt/test_binary_sensor.py

```diff
@@ -872,3 +873,51 @@ async def test_reloadable(hass, mqtt_mock, caplog, tmp_path):
     domain = binary_sensor.DOMAIN
     config = DEFAULT_CONFIG[domain]
     await help_test_reloadable(hass, mqtt_mock, caplog, tmp_path, domain, config)
+
+
+async def test_cleanup_triggers_and_restoring_state(hass, mqtt_mock, caplog, tmp_path):
```

Comment:
We could use `@pytest.mark.freeze_time` + the `freezer` fixture to fake time passing between state change and the reload.